src/node/wsRouter.ts

```diff
@@ -0,0 +1,57 @@
+import * as express from "express"
+import * as expressCore from "express-serve-static-core"
+import * as http from "http"
+import * as net from "net"
+
+export const handleUpgrade = (app: express.Express, server: http.Server): void => {
+  server.on("upgrade", (req, socket, head) => {
+    socket.on("error", () => socket.destroy())
```

Comment:
Perhaps we destroy the socket with the error passed in? I'm surprised node doesn't do this automatically wtf.

Reply:
Actually, it does! Any error will cause the socket to become closed.

https://nodejs.org/api/net.html#net_event_error_1

> Emitted when an error occurs. **The 'close' event will be called directly following this event.**

Reply:
Oh dope! Will remove.